cmd/argocd/commands/headless/headless.go

```diff
@@ -109,6 +117,86 @@ func (c *forwardCacheClient) NotifyUpdated(key string) error {
+	panic("implement me")
+}
+
+func (c *embeddedRepoClientset) TestRepository(ctx context.Context, in *repoapiclient.TestRepositoryRequest, opts ...grpc.CallOption) (*repoapiclient.TestRepositoryResponse, error) {
+	return c.service.TestRepository(ctx, in)
+}
+
+func (c *embeddedRepoClientset) ResolveRevision(ctx context.Context, in *repoapiclient.ResolveRevisionRequest, opts ...grpc.CallOption) (*repoapiclient.ResolveRevisionResponse, error) {
+	return c.service.ResolveRevision(ctx, in)
+}
+
+func (c *embeddedRepoClientset) ListRefs(ctx context.Context, in *repoapiclient.ListRefsRequest, opts ...grpc.CallOption) (*repoapiclient.Refs, error) {
+	return c.service.ListRefs(ctx, in)
+}
+
+func (c *embeddedRepoClientset) ListApps(ctx context.Context, in *repoapiclient.ListAppsRequest, opts ...grpc.CallOption) (*repoapiclient.AppList, error) {
+	return c.service.ListApps(ctx, in)
+}
+
+func (c *embeddedRepoClientset) ListPlugins(ctx context.Context, in *emptypb.Empty, opts ...grpc.CallOption) (*repoapiclient.PluginList, error) {
+	return c.service.ListPlugins(ctx, in)
+}
+
+func (c *embeddedRepoClientset) GetAppDetails(ctx context.Context, in *repoapiclient.RepoServerAppDetailsQuery, opts ...grpc.CallOption) (*repoapiclient.RepoAppDetailsResponse, error) {
+	return c.service.GetAppDetails(ctx, in)
+}
+
+func (c *embeddedRepoClientset) GetRevisionMetadata(ctx context.Context, in *repoapiclient.RepoServerRevisionMetadataRequest, opts ...grpc.CallOption) (*v1alpha1.RevisionMetadata, error) {
+	return c.service.GetRevisionMetadata(ctx, in)
+}
+
+func (c *embeddedRepoClientset) GetRevisionChartDetails(ctx context.Context, in *repoapiclient.RepoServerRevisionChartDetailsRequest, opts ...grpc.CallOption) (*v1alpha1.ChartDetails, error) {
+	return c.service.GetRevisionChartDetails(ctx, in)
+}
+
+func (c *embeddedRepoClientset) GetHelmCharts(ctx context.Context, in *repoapiclient.HelmChartsRequest, opts ...grpc.CallOption) (*repoapiclient.HelmChartsResponse, error) {
+	return c.service.GetHelmCharts(ctx, in)
+}
+
+func (c *embeddedRepoClientset) GetGitFiles(ctx context.Context, in *repoapiclient.GitFilesRequest, opts ...grpc.CallOption) (*repoapiclient.GitFilesResponse, error) {
+	return c.service.GetGitFiles(ctx, in)
+}
+
+func (c *embeddedRepoClientset) GetGitDirectories(ctx context.Context, in *repoapiclient.GitDirectoriesRequest, opts ...grpc.CallOption) (*repoapiclient.GitDirectoriesResponse, error) {
+	return c.service.GetGitDirectories(ctx, in)
+}
+
+func (c *embeddedRepoClientset) UpdateRevisionForPaths(ctx context.Context, in *repoapiclient.UpdateRevisionForPathsRequest, opts ...grpc.CallOption) (*repoapiclient.UpdateRevisionForPathsResponse, error) {
+	return c.service.UpdateRevisionForPaths(ctx, in)
+}
+
+func (c *embeddedRepoClientset) NewRepoServerClient() (io.Closer, repoapiclient.RepoServerServiceClient, error) {
+	return io.NopCloser, c, nil
+}
+
+func newEmbeddedRepoClientset() repoapiclient.Clientset {
```

Comment:
i think we need to pass opts here as well, for future reusability.

Reply:
Moving embedded repo server to another PR